## Data
```
> library("phyloseq"); packageVersion("phyloseq")
[1] ‘1.20.0’
```
### import_qiime
```r
otufile = system.file("extdata", "GP_otu_table_rand_short.txt.gz", package="phyloseq")
mapfile = system.file("extdata", "master_map.txt", package="phyloseq")
trefile = system.file("extdata", "GP_tree_rand_short.newick.gz", package="phyloseq")
rs_file = system.file("extdata", "qiime500-refseq.fasta", package="phyloseq")
qiimedata = import_qiime(otufile, mapfile, trefile, rs_file)
```
```r
> qiimedata
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 500 taxa and 26 samples ]
sample_data() Sample Data:       [ 26 samples by 7 sample variables ]
tax_table()   Taxonomy Table:    [ 500 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 500 tips and 499 internal nodes ]
refseq()      DNAStringSet:      [ 500 reference sequences ]
```
### import_biom
```r
rich_dense_biom  = system.file("extdata", "rich_dense_otu_table.biom",  package="phyloseq")
treefilename = system.file("extdata", "biom-tree.phy",  package="phyloseq")
refseqfilename = system.file("extdata", "biom-refseq.fasta",  package="phyloseq")
import_biom(rich_dense_biom, treefilename, refseqfilename, parseFunction=parse_taxonomy_greengenes)
```
